models/bilibili_api_zh/asr_api.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*
+'''
+    一句话识别 python SDK
+'''
+import sys, json, time
+import codecs
+import base64
+import requests
+URL = "https://asr-speechio.bilibili.com/asr"
+
+
+def is_alnum(c):
+    """
+    简单判断字符是否是英文字母数字
+    """
+    return c.isdigit() or 'a' <= c <= 'z' or 'A' <= c <= 'Z'
+
+
+def trim(s):
+    res = ''
+    for i in range(len(s)):
+        if s[i] == ' ':
+            if 0 < i < len(s) and (is_alnum(s[i - 1]) or is_alnum(s[i + 1])):
+                res += s[i]
+        else:
+            res += s[i]
+    return res
+
+with open('TOKEN', 'r') as f:
+    token = f.readline().strip()
+
+def recognize(audio, key):
+    text = ''
+    for i in range(10):
+        try:
+            with open(audio, "rb") as f:
+                data = f.read()
+                base64Wav = base64.b64encode(data)
+                base64Wav = base64Wav.decode()
+                param = {"wav": base64Wav, 
+                        "key": key, 
+                        "token": token}
+                headers = {"Content-Type": 'application/json'}
+                rep = requests.post(url=URL, data=json.dumps(param), headers=headers)
+                text = trim(json.loads(rep.text)["text"])
```

Comment:
Trimming here seems weird to me and kind of ad-hoc, could you explain why there is such post-processing stage outside bilibili API?